src/settings/configuration.rs

```diff
@@ -28,35 +28,36 @@ use crate::settings::types::{
 
 #[derive(Debug, Default)]
 pub struct Configuration {
+    pub select: Option<Vec<RuleSelector>>,
+    pub ignore: Option<Vec<RuleSelector>>,
+    pub extend_select: Vec<Vec<RuleSelector>>,
+    pub extend_ignore: Vec<Vec<RuleSelector>>,
+    pub fixable: Option<Vec<RuleSelector>>,
+    pub unfixable: Option<Vec<RuleSelector>>,
+    pub per_file_ignores: Option<Vec<PerFileIgnore>>,
```

Comment:
Why pull these up, rather than retain the sort? Semantically, what's the grouping here?

Reply:
These settings control which rules are enabled ... the others don't. Except `per_file_ignores` all of these fields become a `RuleTable` in the `Settings` struct.

Reply:
Sounds good. It might be nice to have a comment like `// Rule enablement` or similar, and something else for the next section, even just to show that the distinction is intentional and not overlooked formatting.